hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/ErrorTableUtils.java

```diff
@@ -63,4 +70,27 @@ public static HoodieErrorTableConfig.ErrorWriteFailureStrategy getErrorWriteFail
     String writeFailureStrategy = props.getString(ERROR_TABLE_WRITE_FAILURE_STRATEGY.key());
     return HoodieErrorTableConfig.ErrorWriteFailureStrategy.valueOf(writeFailureStrategy);
   }
+
+  /**
+   * validates for constraints on ErrorRecordColumn when ErrorTable enabled configs are set.
+   * @param dataset
+   */
+  public static void validate(Dataset<Row> dataset) {
+    if (!isErrorTableCorruptRecordColumnPresent(dataset)) {
+      throw new HoodieValidationException(String.format("Invalid condition, columnName=%s "
+              + "is not present in transformer " + "output schema", ERROR_TABLE_CURRUPT_RECORD_COL_NAME));
+    }
+  }
+
+  public static Dataset<Row> addNullValueErrorTableCorruptRecordColumn(Dataset<Row> dataset) {
+    if (!isErrorTableCorruptRecordColumnPresent(dataset)) {
+      dataset = dataset.withColumn(ERROR_TABLE_CURRUPT_RECORD_COL_NAME, lit(null));
+    }
+    return dataset;
+  }
+
+  private static boolean isErrorTableCorruptRecordColumnPresent(Dataset<Row> dataset) {
+    return Arrays.stream(dataset.columns()).collect(Collectors.toList())
```

Comment:
can we simplify this
```
df.columns.contains(ERROR_TABLE_CURRUPT_RECORD_COL_NAME)
```

Reply:
df.columns is `String[]` , does not have contains method .

Reply:
I guess the intention is to avoid `collect`? Can we simply do `Arrays.stream(dataset.columns()).anyMatch(col -> col.equalsIgnoreCase(ERROR_TABLE_CURRUPT_RECORD_COL_NAME))`